analyzers/src/SonarAnalyzer.Common/Rules/DoNotCheckZeroSizeCollectionBase.cs

```diff
@@ -19,124 +19,123 @@
  */
 
 using System;
+using System.Collections.Generic;
 using System.Collections.Immutable;
+using System.Linq;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Diagnostics;
 using SonarAnalyzer.Helpers;
 
 namespace SonarAnalyzer.Rules
 {
-    public abstract class DoNotCheckZeroSizeCollectionBase<TSyntaxKind, TBinaryExpressionSyntax, TExpressionSyntax> : SonarDiagnosticAnalyzer
+    public abstract class DoNotCheckZeroSizeCollectionBase<TSyntaxKind> : SonarDiagnosticAnalyzer
         where TSyntaxKind : struct
-        where TBinaryExpressionSyntax : SyntaxNode
-        where TExpressionSyntax : SyntaxNode
     {
         protected const string DiagnosticId = "S3981";
-        private const string MessageFormat = "The '{0}' of '{1}' is always '>=0', so fix this test to get the real expected behavior.";
-
+        private const string MessageFormat = "The '{0}' of '{1}' always evaluates as '{2}' regardless the size.";
```

Comment:
Please update `\analyzers\its\expected\Net5\Net5--net5.0-S3981.json` file with the new message to fix ITs from
```
"message":  "The 'Count' of 'IEnumerable<T>' is always '>=0', so fix this test to get the real expected 
```
to
```
"message":  "The 'Count' of 'IEnumerable<T>' always evaluates as 'True' regardless the size.",
```